community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/HeapByteArray.java

```diff
@@ -58,10 +59,30 @@ public void swap( long fromIndex, long toIndex, int numberOfEntries )
     @Override
     public void clear()
     {
-        for ( int i = 0; i < length; i++ )
+        if ( isUniform( defaultValue ) )
         {
-            System.arraycopy( defaultValue, 0, array, i * itemSize, itemSize );
+            Arrays.fill( array, defaultValue[0] );
         }
+        else
+        {
+            for ( int i = 0; i < length; i++ )
+            {
+                System.arraycopy( defaultValue, 0, array, i * itemSize, itemSize );
+            }
+        }
+    }
+
+    private static boolean isUniform( byte[] value )
```

Comment:
we can evaluate uniformity of default value only once instead of doing that each time on clear